### PR TITLE
Make the lightmap size hint property appear as read-only in the editor

### LINKED ISSUE #106698: MeshInstance3D Lightmap size hint not retaining on save
### Tested versions

-Reproducible in main 4.5.dev

Image

Image

### System information

Windows 11 Godot 4.5-Dev Compatibility Renderer 

### Issue description

MeshInstance3D BoxMesh does not retain lightmass hint size after save and scene reload. 

### Steps to reproduce

Add MeshInstance3D set BoxMesh object scale to desired size
Add UV2 and unwrap (check UV2)
Set the hint size to 1024 x 1024 or 64 x 64
Image

### Minimal reproduction project (MRP)

MRP - https://github.com/[uno1982/gitLightBakeMRP](https://github.com/uno1982/gitLightBakeMRP)

### PR DESCRIPTION
**Pull request by Calinou**
Lightmap Size Hint is set on import or when modifying a PrimitiveMesh's size, so manual changes to it are often lost.

However, being able to see it for debugging purposes is useful, so it's visible as a read-only property. This does not impact your ability to set `lightmap_size_hint` from a script (for procedural generation).

- This closes https://github.com/godotengine/godot/issues/106698.
- See discussion starting from https://github.com/godotengine/godot/pull/75164#issuecomment-1478876239.

## Preview

image

### CODE REVIEW

**Comment by Calinou** [comment]
I've improved documentation for the `Mesh.lightmap_size_hint` property.

**Comment by BlueCube3310** [comment]
Since #75164 the user has more control over an individual mesh instance's lightmap size. I think setting the hint to read-only makes more sense now

**Comment by michaelharmonart** [comment]
Although the lightmap size multiplier is nice, there are still use-cases for manually defining the lightmap size of an object in real units (pixels in this case)
https://github.com/godotengine/godot-proposals/issues/11383

I wrote up the above proposal earlier this week specifically because I was running into the the issue of lightmap size hints being reset.

BUT we still need a way to generate better sane defaults for the lightmap hint on imported scenes based on mesh size. Currently large meshes that are impractical to run through Godot's UV2 generation are assigned stupidly small lightmap size hints that mean now you'd be turning the lightmap size scale not to 2x or even 4x but probably on the order of 32x to 64x in many cases. 

Basically my point is: although the lightmap size hint SHOULDN'T need to be edited manually, and will often be reset, for me specifically it's the only way I've used successfully so far to get the result I need (actual decent texel resolution for large chunks of a landscape that have UV2s authored externally)

If we could get lightmap size hints to be set automatically based on the size of imported meshes that aren't run through the godot UV generation, that would be ideal. At that point I'd be 100% on board with this change.

**Comment by uno1982** [comment]
> Basically my point is: although the lightmap size hint SHOULDN'T need to be edited manually, and will often be reset, for me specifically it's the only way I've used successfully so far to get the result I need (actual decent texel resolution for large chunks of a landscape that have UV2s authored externally)
> 
> If we could get lightmap size hints to be set automatically based on the size of imported meshes that aren't run through the godot UV generation, that would be ideal. At that point I'd be 100% on board with this change.

Same... I found it useful to be able to set individual textile density per object as the global setting would globally increase my lightmap disk and vram consumption :(
